### DISCUSSION TITLE
Nodes only scoping to the mesh

### DISCUSSION THREAD

**Discussion by sergo8** [ansys/pydpf-core#1177] [General]
It seems that input parameter: nodes_only=True - does not work.

See the code below:
**node_mesh = core.operators.mesh.from_scoping(scoping=el_scoping, mesh=mesh, nodes_only=True)
node_mesh.eval()
result_mesh = node_mesh.outputs.mesh()**

It gives the following output: 

**DPF  Meshed Region: 
  668 nodes 
  75 elements 
  Unit: mm 
  With solid (3D) elements**
  
  Instead, I guess it has to return only selected nodes, but not both elements and nodes.
  
Is this a bug or I do it in a wrong way?

**Comment by mikerife** [comment]
Hi @sergo8 
That looks right.  I just tried this with PyDPF version 0.9

```python
from ansys.dpf import core as dpf
from ansys.dpf.core import examples

model = dpf.Model(examples.find_static_rst())
mesh = model.metadata.meshed_region

my_element_scoping = dpf.mesh_scoping_factory.elemental_scoping([1])
node_mesh = dpf.operators.mesh.from_scoping(scoping=my_element_scoping, mesh=mesh, nodes_only=True)
node_mesh.eval()
result_mesh = node_mesh.outputs.mesh()
print(result_mesh)
```
And it returned:

![image](https://github.com/ansys/pydpf-core/assets/91209185/f7a3f9d2-eaef-4a56-b866-365921d70d76)

Is that what you had in mind?  
Mike

**Reply by sergo8** [reply]
Hi @mikerife,

I am getting this result below with the same code as you have:

![image](https://github.com/ansys/pydpf-core/assets/26027408/47784ee6-92b8-4616-a96b-e593bf6750ed)

Version is 0.9.

from ansys.dpf import core
from ansys.dpf.core import mesh_scoping_factory

rst_file = r'..\dp0\SYS\MECH\file.rst'
model = core.Model(rst_file)

mesh = model.metadata.meshed_region
el_scoping = mesh_scoping_factory.elemental_scoping([1])

node_mesh = core.operators.mesh.from_scoping(scoping=el_scoping, mesh=mesh, nodes_only=True)
node_mesh.eval()

result_mesh = node_mesh.outputs.mesh()
print(result_mesh)

**Reply by mikerife** [reply]
@sergo8 Ok...looks like it's time to open an Issue if you can provide the rst file.  If you cannot provide the rst file can you explain in detail the features of the model so we can try to replicate?
Mike

**Reply by sergo8** [reply]
@mikerife 
The .rst file exceeds 25 Mb, I can't attach it to the message.
But it is a simple model. A solid model with hex elements + midside nodes that has a named selection on the edge. I am trying to select all nodes that belong to the named selection. But, instead it selects elements + nodes that belong to this named selection. 
This is basically the issue. I can't even select a single node using your code provided above, it still selects element + nodes.

**Reply by mikerife** [reply]
@sergo8 Then you will need to create a smaller test model.  

Can you post a screen shot of the named selection?  An edge NS (in WB Mechanical) is a list of nodes, not elements.

**Reply by sergo8** [reply]
@mikerife 
![image](https://github.com/ansys/pydpf-core/assets/26027408/4aa04802-5844-431b-929e-a72c88ebca62)

I need to select these nodes, but instead, it selects all neighbor elements that share the same nodes + nodes of the named selection. On top of nodes  that already in the named selection it selects all nodes that belong to these elements as well.

See the full code below:
```
from ansys.dpf import core
from ansys.dpf.core import mesh_scoping_factory
import numpy as np

rst_file = r'..\dp0\SYS\MECH\file.rst'
model = core.Model(rst_file)

mesh = model.metadata.meshed_region

named_selection_1 = "WELD_1_N"
named_selection_2 = ""

scope_ns1 = mesh_scoping_factory.named_selection_scoping(named_selection_1, model)
scope_ns2 = mesh_scoping_factory.named_selection_scoping(named_selection_2, model)

el_scoping = core.Scoping()
el_scoping.location = core.locations.nodal
new_list_scoped_elements = np.concatenate((scope_ns1.ids, scope_ns2.ids))

el_scoping.ids = new_list_scoped_elements

node_mesh = core.operators.mesh.from_scoping(scoping=el_scoping, mesh=mesh, nodes_only=True)
node_mesh.eval()

result_mesh = node_mesh.outputs.mesh()
print(result_mesh)
```

![image](https://github.com/ansys/pydpf-core/assets/26027408/8e465b33-09b2-4206-9084-89637f604faa)
